include/envoy/thread/thread.h

```diff
@@ -10,15 +12,23 @@
 namespace Envoy {
 namespace Thread {
 
+/**
+ * An id for a thread.
+ */
 class ThreadId {
 public:
-  virtual ~ThreadId() = default;
+  ThreadId() : id_(std::numeric_limits<int64_t>::min()) {}
+  ThreadId(int64_t id) : id_(id) {}
+  ~ThreadId() = default;
 
-  virtual std::string debugString() const PURE;
-  virtual bool isCurrentThreadId() const PURE;
-};
+  std::string debugString() const { return std::to_string(id_); }
+  bool isEmpty() const { return *this == ThreadId(); }
+  friend bool operator==(ThreadId lhs, ThreadId rhs) { return lhs.id_ == rhs.id_; }
+  friend bool operator!=(ThreadId lhs, ThreadId rhs) { return lhs.id_ != rhs.id_; }
```

Comment:
I thought one of the goals of this PR was to make this hashable, but I don't see a way to get a hash out yet.

Reply:
it was to make it easy for it to be added ;)  I will add it in a few days if you want

Reply:
added!